cedar-policy-validator/src/diagnostics/validation_errors.rs

```diff
@@ -448,6 +449,114 @@ impl Diagnostic for HierarchyNotRespected {
     }
 }
 
+#[derive(Debug, Clone, Hash, Eq, PartialEq, Error, Copy)]
+/// Represents how many entity dereferences can be applied to a node.
+/// `None` represents infinity.
+pub struct EntityDerefLevel {
+    /// `None` represents infinity.
+    pub level: Option<i64>,
```

Comment:
Why signed?

Reply:
It's for error messages. Actual level is `allowed_level.add(actual_level.neg())`